syntax = "proto3";

package grpc;

message Empty {}

message User {
    int64 uuid = 1;
    string username = 2;
    string password = 3;
}

message Chat {
    string username = 3;
    string message = 4;
}

message Outcome {
    int64 err_type = 1;
    string err_msg = 2;
}

service ChatBot {
    rpc server_send_chat (Chat) returns (Outcome);
    rpc server_get_chat (User) returns (Chat); 
}